### PR TITLE
pkg/identity: Add basic identity allocator

### PR DESCRIPTION
**Pull request by ovidiutirla**
Basic identity allocator will be used by operator to manage global identities CID and agent to manage locally created identities that do not require complex features like `pkg/allocator/allocator.go`.

Related: #30356
Draft full implementation: https://github.com/cilium/cilium/pull/33204

```release-note
Add basic identity allocator which will be used by operator to manage global CID and agent to manage local identities.
```

### CODE REVIEW

**Comment by ovidiutirla** [comment]
This is a follow-up on @dlapcevic work for Operator Managing CIDs, please have another look over this.

/cc @squeed @joestringer @marseel

**Review by dlapcevic — APPROVED** [review]
lgtm

**Comment by dlapcevic** [comment]
Hi @derailed, could we please get a review for this?

File: pkg/identity/basicallocator/basic_allocator.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// the provided minIDValue and maxIDValue.
+func NewBasicIDAllocator(minIDValue, maxIDValue idpool.ID) *BasicIDAllocator {
+	return &BasicIDAllocator{
+		idPool:     idpool.NewIDPool(minIDValue, maxIDValue),
+		minIDValue: minIDValue,
+		maxIDValue: maxIDValue,
+	}
+}
+
+func (b *BasicIDAllocator) AllocateRandom() (idpool.ID, error) {
```
Comment (derailed) [inline pkg/identity/basicallocator/basic_allocator.go:35]: Exported method here and below should have docs.

**Review by derailed — COMMENTED** [review]
@ovidiutirla Nice. Thank you for these udpates!

File: pkg/identity/basicallocator/basic_allocator.go (hunk 1 above)
Comment (ovidiutirla) [reply pkg/identity/basicallocator/basic_allocator.go:35]: Hey @derailed,
I've updated the docs.

**Comment by joestringer** [comment]
(Please continue review etc as usual, but I'm deferring merge until after we branch for v1.16 in the next week or so, since this is a start of a new effort that is not targeted for v1.16).

**Comment by ovidiutirla** [comment]
rebased on main

**Comment by pchaigno** [comment]
@ovidiutirla You'll need to rebase again to catch https://github.com/cilium/cilium/pull/33572 and fix the CI failures.

**Review by pchaigno — APPROVED** [review]

**Review by sayboras — APPROVED** [review]
Assume that this is the first steps only, looking forward for your next set of changes.

**Comment by dlapcevic** [comment]
🎉

**Pull request merged**